themes/src/main/resources/theme/keycloak-preview/account/resources/app/keycloak-service/keycloak.service.ts

```diff
@@ -84,6 +84,21 @@ export class KeycloakService {
         return KeycloakService.keycloakAuth.realm;
     }
 
+    public get userName(): string {
+        let userName = 'Anonymous';
+        if (KeycloakService.keycloakAuth.tokenParsed) {
+            const givenName = (KeycloakService.keycloakAuth.tokenParsed as {given_name: string}).given_name;
+            const familyName = (KeycloakService.keycloakAuth.tokenParsed as {family_name: string}).family_name;
+            const preferredUsername = (KeycloakService.keycloakAuth.tokenParsed as {preferred_username: string}).preferred_username;
+            if (givenName && familyName) {
+                userName = givenName + ' ' + familyName;
```

Comment:
This needs to be localized as well.  Many cultures don't do givenName + ' ' + familyName.  So these should be passed in as params to a message bundle instead.

Reply:
givenName (`given_name`) and familyName (`family_name`) are the properties in `tokenParsed` so what do you want me to localize? The way we concatenate the name? Are there more properties in `tokenParsed` that contain the parts of the name?

Reply:
Yes, the way we concatenate the name should be localized.  I think all you need to do is pass the params into Msg.  Then the message bundle concatenates based on locale.  So English bundle looks something like:
fullName={0} {1}

South Korea bundle might look like:
fullName={1} {0}

Reply:
okay done